### PR TITLE
Bugfixes: Default nfft value and to_snake_case implementation

### PR DESCRIPTION
**Pull request by jantonguirao**
#### Why we need this PR?
- It fixes a bugs found during NeMo integration

Thanks to @mzient for to_snake_case implementation

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     Fixed two bugs:
       1. Taking nfft from window_len when the former is not provided.
       2. Fix to_snake_case conversion ("MFCC" was converted to "mfc" instead of "mfcc"
 - Affected modules and functionalities:
     *dali.fn and ops.Spectrogram*
 - Key points relevant for the review:
     *N/A*
 - Validation and testing:
     *Tests added*
 - Documentation (including examples):
     *N/A*

**JIRA TASK**: *[DALI-1556]*

### CODE REVIEW

File: dali/operators/signal/fft/spectrogram_gpu.cc [hunk 1]
```diff
@@ -34,7 +34,7 @@ struct SpectrogramOpImplGPU : public OpImplBase<GPUBackend> {
   explicit SpectrogramOpImplGPU(const OpSpec &spec) {
     args.window_length = spec.GetArgument<int>("window_length");
     args.window_step = spec.GetArgument<int>("window_step");
-    args.nfft = spec.GetArgument<int>("nfft");
+    args.nfft = spec.HasArgument("nfft") ? spec.GetArgument<int>("nfft") : args.window_step;
```
Comment (mzient) [inline dali/operators/signal/fft/spectrogram_gpu.cc:37]:
```suggestion
    args.nfft = spec.HasArgument("nfft") ? spec.GetArgument<int>("nfft") : args.window_length;
```

**Review by JanuszL — APPROVED** [review]

**Review by mzient — APPROVED** [review]

File: dali/test/python/test_operator_fn.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        ('OneXYZTwo', 'one_xyz_two'),
+        ('MFCC', 'mfcc'),
+        ('RandomBBoxCrop', 'random_bbox_crop'),
+        ('STFT_CPU', 'stft_cpu'),
+        ('DOUBLE__UNDERSCORE', 'double__underscore'),
+        ('double__underscore', 'double__underscore'),
+        ('XYZ1ABC', 'xyz1abc'),
+        ('XYZ1abc', 'xyz1abc'),
+        ('trailing__', 'trailing__'),
+        ('TRAILING__', 'trailing__')
```
Comment (szalpal) [inline dali/test/python/test_operator_fn.py]: Additional names worth adding:
```
Caffe2Reader
COCOReader
DLTensorPythonFunction
TFRecordReader
```

Comment (jantonguirao) [reply dali/test/python/test_operator_fn.py]: Done

**Comment by jantonguirao** [comment]
!build

File: dali/test/python/test_operator_fn.py (hunk 2 above)
Comment (mzient) [reply dali/test/python/test_operator_fn.py]: :+1: for TFRecord, as it's a special case, along with BBox and MXNet.

**Pull request merged**